docs/step_by_step_guide.md

```diff
@@ -0,0 +1,646 @@
+- `readonly DEBUG_COMPILE=false`: option to compile with DEBUG flag (leave set to false)
+
+> [!IMPORTANT]
+> BEFORE RUNNING: Change `CHECKOUT` to a date string like 20240301. 
+
+> [!IMPORTANT]
+> A case is tied to one code base and one executable. That is, if you change `CHECKOUT` or `BRANCH`, then you should also change `CASE_NAME`.
+
+`# Run options`
+- `readonly MODEL_START_TYPE="initial"`: specify how the model should start – use initial conditions,  continue from existing restart files, branch, or hybrid (respectively: initial, continue, branch, hybrid).
+- `readonly START_DATE="0001-01-01"`: model start date. Typically year 1 for simulations with perpetual (time invariant) forcing or a real year for simulations with transient forcings.
+
+`# Set paths`
+- `readonly CODE_ROOT="${HOME}/E3SMv3/code/${CHECKOUT}"`: where the E3SM code will be checked out.
+- `readonly CASE_ROOT="/pscratch/sd/r/${USER}/e3sm-scratch/${CASE_NAME}"`: where the results will go. The directory will be `<simulations_dir>/${CASE_NAME}`.
+
+`# Sub-directories`
+- `readonly CASE_BUILD_DIR=${CASE_ROOT}/build`: all the compilation files, including the executable.
+- `readonly CASE_ARCHIVE_DIR=${CASE_ROOT}/archive`: where short-term archived files will reside.
+
+`# Define type of run`
+- `readonly run='XS_2x5_ndays'`: type of simulation to run – i.e, a short test for verification or a long production run. (See next section for details).
+
+`# Coupler history`
+- `readonly HIST_OPTION="nyears"`
+- `readonly HIST_N="5"`
+
+`# Leave empty (unless you understand what it does)`
+- `readonly OLD_EXECUTABLE=""`: this is a somewhat risky option that allows you to re-use a pre-existing executable. This is not recommended because it breaks provenance.
+
+`# --- Toggle flags for what to do ----`
+
+This section controls what operations the script should perform. The run_e3sm script can be invoked multiple times with the user having the option to bypass certain steps by toggling true / false.
+- `do_fetch_code=true`: fetch the source code from Github.
+- `do_create_newcase=true`: create new case.
+- `do_case_setup=true`: case setup.
+- `do_case_build=false`: compile.
+- `do_case_submit=true`: submit simulation.
+
+The first time the script is called, all the flags should be set to true. Subsequently, the user may decide to bypass code checkout (`do_fetch_code=false`) or compilation (`do_case_build=false`). A user may also prefer to manually submit the job by setting `do_case_submit=false` and then invoking `./case.submit`.
+
+# Running the Model
+
+## Short Tests
+Before starting a long production run, it is *highly recommended* to perform a few short tests to verify:
+1. The model starts without errors.
+2. The model produces BFB (bit-for-bit) results after a restart.
+3. The model produces BFB results when changing PE layout.
```

Comment:
The numbered list works in the GitHub rendering, but not in the actual rendering (where it shows up all in one line).